lib/rouge/lexers/supercollider.rb

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class SuperCollider < RegexLexer
+      tag 'supercollider'
+      filenames '*.sc', '*.scd'
+
+      title "SuperCollider"
+      desc 'A cross-platform interpreted programming language for sound synthesis, algorithmic composition, and realtime performance'
+
+      keywords = Set.new %w(
+        var arg classvar const super this
+      )
+
+      # these aren't technically keywords, but we treat
+      # them as such because it makes things clearer 99%
+      # of the time
+      reserved = Set.new %w(
+        case do for forBy loop if while new newCopyArgs
+      )
+
+      constants = Set.new %w(
+        true false nil inf thisThread
+        thisMethod thisFunction thisProcess
+        thisFunctionDef currentEnvironment
+        topEnvironment
+      )
+
+      start { push :bol }
+
+      # beginning of line
+      state :bol do
+        mixin :inline_whitespace
+
+        rule(//) { pop! }
+      end
+
+      state :inline_whitespace do
+        rule /\s+/m, Text
+        mixin :has_comments
+      end
+
+      state :whitespace do
+        rule /\n+/m, Text, :bol
```

Comment:
You shouldn't need to return to the beginning of the line (really the beginning of the file) state once you leave it. Is there a reason you're doing it this way?